client/src/components/admin_announcement/AddAnnouncement.tsx

```diff
@@ -0,0 +1,272 @@
+import draftToHtml from "draftjs-to-html";
+
+import { useHistory } from "react-router-dom";
+
+import { Button } from "../button/Button";
+
+import { AnnouncementInput } from "../../types/announcement";
+
+interface AddAnnouncementProps {
+  className?: string;
+  onAdd: (announcement: AnnouncementInput) => Promise<void>;
+}
+
+/**
+ * Announcement.
+ */
+const UnstyledAddAnnouncement: FunctionComponent<AddAnnouncementProps> = ({
+  className,
+  onAdd,
+}) => {
+  const [editorState, setEditorState] = useState<EditorState>(
+    EditorState.createEmpty()
+  );
+
+  const [newAnnouncement, setNewAnnouncement] = useState<AnnouncementInput>({
+    title: "",
+    subtitle: "",
+    content: "",
+    showOnHomepage: false,
+    images: [],
+  });
+
+  const history = useHistory();
+
+  const isValid = useMemo(
+    () =>
+      !!newAnnouncement?.title &&
+      !!newAnnouncement?.subtitle &&
+      !!newAnnouncement?.content,
+    [newAnnouncement]
+  );
+
+  useEffect(
+    () =>
+      setNewAnnouncement({
+        title: "",
+        subtitle: "",
+        content: "",
+        showOnHomepage: false,
+      }),
+    []
+  );
+
+  const uploadImageCallBack = (file: File) => {
+    const imageObject = {
+      file: file,
+      localSrc: URL.createObjectURL(file),
+    };
+
+    return new Promise((resolve, reject) => {
+      resolve({ data: { link: imageObject.localSrc } });
+    });
+  };
+
+  return (
+    <Box className={className}>
+      <Box display="flex">
+        <Paper className="announcement-paper">
+          <Box className="announcement-title">
+            <Typography variant="h5">Announcement Form</Typography>
+          </Box>
+
+          <Box className="form-fields" display="flex">
+            <Typography> Title* </Typography>
+            <TextField
+              className="title-textfield"
+              fullWidth
+              variant="outlined"
+              color="secondary"
+              value={newAnnouncement?.title}
+              onChange={(evt: ChangeEvent<HTMLInputElement>) => {
+                setNewAnnouncement({
+                  ...newAnnouncement,
+                  title: evt.target.value,
+                });
+              }}
+            ></TextField>
+          </Box>
+
+          <Box className="form-fields" display="flex">
+            <Typography> Subtitle* </Typography>
+            <TextField
+              className="subtitle-textfield"
+              fullWidth
+              variant="outlined"
+              color="secondary"
+              value={newAnnouncement?.subtitle}
+              onChange={(evt: ChangeEvent<HTMLInputElement>) => {
+                setNewAnnouncement({
+                  ...newAnnouncement,
+                  subtitle: evt.target.value,
+                });
+              }}
+            ></TextField>
+          </Box>
+
+          <Box className="form-fields">
+            <Typography> Body Content* </Typography>
+
+            <Editor
```

Comment:
This should be factored out into wrapper component. We talked about this

Reply:
Fixed the error, just need to refactor DraftEditor part and make it work.